### PR TITLE
fix(wspr): match WSJT-X on tone convention, frame taper and slot timing

### PR DESCRIPTION
**Pull request by jensenpat**
## What

The WSPR beacon transmitted a frame that decoded, but it differed from the
reference implementation in four ways that each cost signal quality or
accuracy. WSJT-X's `Modulator.cpp` is the oracle throughout — a WSPR frame is
judged by decoders we do not own, so its conventions are not ours to choose.

The reported symptom was "it opens the band in USB, not DIGU, with a very wide
open filter." That turned out to be real but only one of the five things wrong.

## The tone convention was 2.2 Hz low

WSJT-X puts tone 0 **at** the requested audio frequency and runs the
constellation upward from it — `m_frequency + itone[isym] * m_toneSpacing`,
itone in 0..3. The generator centred the four tones on that value instead, so
every tone sat 1.5 spacings — **2.197 Hz** — below where the same number in
WSJT-X would have put it, and every resulting spot report with it.

## The frame was cut, not faded

`WsprBeacon::process()` stopped emitting at symbol 162 at whatever phase and
amplitude it had reached: a step discontinuity from full scale to zero, which
is a broadband click in a 200 Hz-wide sub-band shared with every other WSPR
station on the band.

WSJT-X fades its tail — `i0 = (m_symbolsLength - 0.017) * 4.0 * m_nsps`, then
`m_amp = 0.98 * m_amp` per sample at 48 kHz, about -98 dB by the final sample.
Matched here: the same 11.6 ms, which at our 24 kHz rate is half as many
samples and so needs the factor squared, **0.9604 over 278 frames**.

The **head** ramp is an addition beyond WSJT-X, which only fades the tail. It
costs 1.7% of one sync symbol and removes the derivative discontinuity at
key-on. Flagging it explicitly as a deliberate deviation rather than burying
it — phase is continuous across symbol boundaries, so these two ends are the
only discontinuities in 111.6 seconds and both are now shaped.

## Timing was referenced to the wrong clock

The lead-in was a flat one second measured from **whenever the pump started**,
so the 50 ms GUI tick, the PTT round trip and the stream start all landed in
DT instead of being absorbed.

It is now referenced to the slot boundary, as WSJT-X does against the wall
clock (`m_silentFrames = (delay_ms - mstr) * m_frameRate / 1000`). Past the
1.000 s mark the remainder truncates the **head** of the frame — WSJT-X's
`m_ic = (mstr - delay_ms) * m_frameRate / 1000` — rather than sliding a
111.6 s transmission later into a two-minute slot. The head ramp follows
frames *emitted* rather than position in the message, so a skipped start is
still ramped.

Residual unmeasured latency is the queued hop to the audio thread, which is
sub-millisecond against a two-second lateness budget.

## The channel was pushed once and never checked

This is the reported bug. `applyBeaconBand()` sent `slice tune`, `mode=DIGU`
and `filt` back to back, and nothing looked again.

A cross-band `slice tune` makes the radio recall freq/mode/filters from its
**band stack**, asynchronously, and it may recreate the slice while doing it
(flexlib-oracle §6.2; #2824). That recall can land *after* the mode and filter
sent immediately behind it, and when it does it wins — leaving the beacon
pointed at a USB slice with whatever passband the band stack held.
`SliceModel::setMode()` also deliberately declines to push a `filt` on the Flex
path, so the mode echo carries a filter of the radio's choosing.

And arming happens up to a **full 120 seconds** before the key. Another client,
a band button or a profile load can move the slice in that window.

`reassertBeaconChannel()` now re-sends mode and both passbands immediately
before `requestPttOn()`. The generated lead-in is silence, so the radio has all
of it to apply them before the first symbol. A dial that **moved** aborts
rather than being re-tuned — a fresh band-stack recall with only the pre-roll
to settle in, on a frequency the operator did not choose, is worse than not
transmitting.

## Nothing neutralized the speech chain

A Flex does not bypass the speech processor, the compander or the TX equalizer
because a slice says DIGU. Left on, they work on a constant-envelope 4-FSK
tone: the compander pumps its level across the frame, the processor adds
intermodulation either side of the carrier, the EQ tilts it. This is exactly
why WSJT-X's operating guidance is to switch all of it off.

Arming now borrows and restores them the same way the TX filter already was,
plus VOX — which is not audio shaping but a second thing that can *unkey*
part-way through a 111.6 s frame. Only what was actually on gets turned back
on, so a restore can never enable something the operator had off. Flex command
plane only.

## Smaller items

- Phase accumulator is `double`, as in WSJT-X, rather than `float`.
- The DAX pump generates float directly instead of quantizing to int16 and
  dividing straight back — an undithered quantization of a pure tone, the one
  signal whose quantization error is harmonically correlated rather than
  noise-like.
- Transmit level is an operator control (-60..-3 dBFS, persisted) instead of
  hard-coded at -20. WSJT-X generates full scale and attenuates digitally
  through its Pwr slider (`SoundOutput::setAttenuation`); the equivalent knob
  here had no UI at all, so an operator who came out underdriven had nothing to
  reach for.

## Verification

**On air, and this is the part that matters.** Two 40 m transmissions ten hours
apart, one per build, as independently reported by wspr.live:

| Transmission | Build | Spots | Median reported |
|---|---|---|---|
| 03:14Z | before | 70 | 7,040,100.0 Hz |
| 13:54Z | after | 61 | 7,040,102.5 Hz |

**+2.5 Hz measured against the +2.197 Hz the tone-convention change predicts**,
by ~60 receivers that have never seen this source. The residual 0.3 Hz is
inside WSPR's own frequency-estimation noise. Within each transmission the
spread across independent receivers is about 1 Hz and `drift` is 0 on the large
majority — a clean, well-formed carrier. Best DX **VK5HW `PF94hk` at
13,237 km** at 37 dBm, with five Australian and two New Zealand receivers.

This matters because a convention error is invisible to any test that shares
the convention. The unit tests below assert that the generator matches *my
reading* of `Modulator.cpp`; only the spot reports confirm the reading.

**Unit tests — 4 new cases, each verified to have teeth by reverting the
behaviour under it:**

- Reverting to the centred tone convention fails both tone assertions. The test
  counts cycles over ten symbols: because one tone step is exactly one cycle
  per symbol, all-zeros is 10240 and all-threes is 10270, where the old
  convention gives 10225 — 15 cycles clear, not a rounding-width margin.
- Forcing the envelope to 1.0 fails all three taper assertions.

Full suite green: **204/204**.

## Not covered

Type-2 (compound call) and type-3 (hashed call) messages, Tx-percentage
scheduling and band hopping, and FST4W are all still absent. Out of scope here
by agreement — this PR is transmit quality for the type-1 frame we already
send.

Selecting a WSPR band still tunes only the dial, leaving mode and filter alone
until arm time. That is #4537's deliberate choice — the operator stays in their
own mode to listen before committing — so it is not overturned here. What is
now guaranteed is that the *transmission* is DIGU at 1200–1800, verified at the
key rather than hoped for at arm.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/gui/PskReporterMapDialog.cpp [hunk 1]
```diff
@@ -719,6 +754,32 @@ bool PskReporterMapDialog::applyBeaconBand()
...
+        m_beaconPrevCompander = tx.dexpOn();
+        m_beaconPrevVox = tx.voxEnable();
+        m_beaconPrevTxEq = eq.txEnabled();
+        m_beaconTxChainSaved = true;
+        if (m_beaconPrevSpeechProc) tx.setSpeechProcessorEnable(false);
+        if (m_beaconPrevCompander) tx.setDexp(false);
+        // VOX is not audio shaping — it is a second thing that can key and
+        // unkey the transmitter. Ours is a 111.6 s frame held by an explicit
+        // MOX, and a VOX release part-way through would truncate it.
+        if (m_beaconPrevVox) tx.setVoxEnable(false);
```
[the remark this answers is not in the document]
Comment (jensenpat) [reply src/gui/PskReporterMapDialog.cpp]: This is dumb, you wouldn't use VOX during a Whisper TX. You would hard or manual abort with the TX indicator.

**Review by ten9876 — COMMENTED** [review]
**Issue fit.** No linked issue (`closingIssuesReferences` is empty), so I reviewed against the PR's own stated intent. The reported symptom — "opens the band in USB, not DIGU, with a very wide open filter" — maps to `reassertBeaconChannel()`, and the four generator changes map to the `Modulator.cpp` conventions the body cites. Per GOVERNANCE.md this lands in "bug fixes with a clear root cause" plus "protocol compliance fixes matching SmartSDR behavior", and the new Level control is additive with the previous hard-coded −20 dBFS as its default, so nothing here owes an RFC.

**What I verified locally**, in a worktree at `15e2397f` merged onto current `main` (27 commits ahead of this PR's base — the merge is clean, and main's `AudioEngine.cpp` churn does not touch the WSPR pump):

- `wspr_beacon_test` builds and passes.
- **Both new tests have teeth, confirmed by mutation, not taken on faith:**
  - restoring the centred tone (`(symbol - 1.5) * spacing`) →
    `FAIL: symbol 0 is emitted at the requested frequency, not below it` and
    `FAIL: symbol 3 sits three tone spacings above the requested frequency`.
  - forcing `m_envelope` to `1.0f` → all three taper assertions fail.
- **Taper arithmetic**: `kTaperFrames` = 16384·17/1000 = 278; the seed `0.9604^278` ≈ −97.6 dB; the ramp reaches unity at emitted frame 277, and the `else` branch pins the envelope to `1.0f` afterwards, so float drift across the 278 divisions cannot leak into the body. The tail is exactly 278 frames.
- **Lead-in arithmetic** checks out in both branches: on time, audio starts at slot + 1.000 s regardless of tick jitter; when `lateMs > 1000`, `skipFrames = (lateMs − 1000)·24` is exactly the message position at wall time slot + lateMs. Head truncation, not a slid frame.
- **The re-assert genuinely re-sends.** Neither `SliceModel::setFilterWidth()` nor `TransmitModel::setTxFilter()` has an equality guard, so both always hit the wire. `SliceModel::setMode()` *does* early-return on `m_mode == mode`, but the status path (`SliceModel.cpp:1068`) mirrors the radio's reported mode back into `m_mode`, so a band-stack recall to USB leaves the model reading USB and the `mode=DIGU` push happens. The fix does what it says.
- `setDexp` → `transmit set compander=` (`TransmitModel.cpp:579-585`, backed by the FlexLib v4.2.18 + pcap citation) — right handle for the compander.
- Settings ride the single `PskReporter` JSON root (Constitution V), and the new spin box carries an accessible name and description.
- The `AudioEngine` float-scratch change is behaviour-equivalent: the buffer is zero-filled before `process()`, so the stop-inside-the-window comment above it still holds.

**No blockers.** One design point that I think is a maintainer call rather than mine, and five nits — all inline, all explicitly non-blocking.

The design point is at `PskReporterMapDialog.cpp:767`: the speech chain is borrowed at *arm* and, unlike mode and the passbands, never looked at again. That is the same "pushed once and never checked" shape this PR exists to remove, left in place for the four settings the PR adds. @aethersdr-agent raised the operator-facing half of this and you pushed back on the VOX argument; the re-check half is separate and I think survives that objection. Both are fixed by the same move, so it's worth one ruling.

For the record: the CodeGuard comment on this PR is noise — all five findings cite `src/gui/MainWindow.cpp`, which this PR does not touch.

File: src/gui/PskReporterMapDialog.cpp [hunk 2]
```diff
@@ -719,6 +754,32 @@ bool PskReporterMapDialog::applyBeaconBand()
...
+        if (m_beaconPrevTxEq) eq.setTxEnabled(false);
```
Comment (ten9876) [inline src/gui/PskReporterMapDialog.cpp]: **Non-blocking, but the one thing I'd want a ruling on before merge.**

This PR's central argument is in `reassertBeaconChannel()`'s own comment: *"One push at arm time is not enough… arming happens up to a full 120 s before the key. Anything at all can move the slice in that window."* That argument is applied to the slice mode and both passbands. It is not applied to the four settings taken right here, in the same function, which are pushed once at arm and never re-checked.

The gap is concrete and independent of the VOX debate below: if the operator (or another client, or a profile load) turns the speech processor or the compander back on during the armed window, **the beacon transmits through it** and the on-air quality fix this section exists for is silently defeated. Nothing looks again. That is the defect class the rest of the PR closes.

The window is also longer than "up to 120 s": with `kBeaconMaxDeferrals` = 2 and the new transmitter-busy deferral you added at line 1115, arm→key can span three slots, so roughly six minutes.

Moving the whole block into `reassertBeaconChannel()` fixes both halves at once and costs nothing you have not already argued for — it is the same lead-in, the same `isTransmitting()/isTuning()` guard, and it makes the *saved* values fresh rather than up to six minutes stale (which matters for the restore: `m_beaconPrevSpeechProc` is latched at arm, so a mid-window change is restored to the wrong value).

On the VOX half specifically — you're right that nobody would run VOX *during* a WSPR transmission. The bot's point was about the *armed* window, where this very PR's new deferral comment (line 1106) says the operator may "MOX for a quick voice contact". A VOX operator in that window finds the mic dead with no cause shown. If you'd rather keep the borrow at arm, the cheap mitigation is a word in `m_beaconStatus` while armed; I don't think it needs more than that. Either way it's your radio and your call — flagging it so it's a decision rather than an oversight.

(For what it's worth, the operator *can* still cancel: `setBeaconControlsEnabled(false)` leaves `m_beaconButton` enabled as "Cancel". The bot's claim that they can't is wrong.)

File: src/core/WsprBeacon.cpp [hunk 3]
```diff
@@ -224,58 +230,102 @@ void WsprBeacon::beginPendingTransmission() noexcept
...
+    const int64_t framePos =
+        static_cast<int64_t>(m_symbolIndex) * kFramesPerSymbol
+        + m_framesIntoSymbol;
+    if (m_framesEmitted < kTaperFrames) {
+        m_envelope = std::min(1.0f, m_envelope / kTaperDecayPerFrame);
+    } else if (framePos >= kMessageFrames - kTaperFrames) {
+        m_envelope *= kTaperDecayPerFrame;
+    } else {
+        m_envelope = 1.0f;
+    }
```
Comment (ten9876) [inline src/core/WsprBeacon.cpp]: Nit, non-blocking — confirming @aethersdr-agent's finding after checking it myself.

The head-ramp branch wins over the tail branch, so a `messageSkipFrames` that lands `framePos` within `kTaperFrames` of the end ramps *up* for the few frames that remain and then stops at whatever level it reached — a step discontinuity, which is the click the taper exists to prevent.

Unreachable from `updateBeaconState()`: `kBeaconMaxSlotLatenessMs` is 2000, so `skipFrames` tops out at 24 000 against `kMessageFrames` = 2 654 208. But `start()` is public, clamps only to `kMessageFrames - 1`, and the new test helper calls it directly.

Swapping the branch order is cheaper than clamping in `start()` and is a no-op on the reachable path (during the ramp, `framePos` is ~24 000, nowhere near the tail region):

```suggestion
    if (framePos >= kMessageFrames - kTaperFrames) {
        m_envelope *= kTaperDecayPerFrame;
    } else if (m_framesEmitted < kTaperFrames) {
        m_envelope = std::min(1.0f, m_envelope / kTaperDecayPerFrame);
    } else {
        m_envelope = 1.0f;
    }
```

With the tail winning, a skip that deep starts at the seed envelope (−97.6 dB) and only decays further, so the frame ends silent instead of clipped. The comment above will want its last sentence adjusted to match.

File: docs/architecture/pipelines.md [hunk 4]
```diff
@@ -107,6 +107,29 @@ independently of microphone callbacks and sends it through a client-owned
...
+  untapered one is a broadband click in a 200 Hz-wide shared sub-band.
+- **The lead-in is referenced to the slot**, not to whenever the pump started,
+  so scheduling latency is absorbed rather than reported as DT.
+- **A missed boundary truncates the head** of the frame instead of sliding the
+  whole transmission later.
+
+Arming also borrows station state that is not slice state and hands it back on
+stop: the transmit passband, the speech processor, the compander and the TX
+equalizer. None of those are bypassed by selecting DIGU, and all of them
+misshape a constant-envelope tone. Mode and both passbands are re-asserted
```
Comment (ten9876) [inline docs/architecture/pipelines.md]: Nit, non-blocking: the list omits VOX, which `applyBeaconBand()` also takes and `restoreBorrowedTxState()` also hands back. Doc drift in the paragraph this PR is adding.

```suggestion
Arming also borrows station state that is not slice state and hands it back on
stop: the transmit passband, the speech processor, the compander, the TX
equalizer and VOX. None of the first four is bypassed by selecting DIGU and all
of them misshape a constant-envelope tone; VOX is taken because it can unkey
part-way through a 111.6 s frame. Mode and both passbands are re-asserted
```

File: tests/wspr_beacon_test.cpp [hunk 5]
```diff
@@ -212,6 +351,9 @@ int main(int argc, char** argv)
     testValidation();
     testSampleTimingAndTailSilence();
     testTwentyOneSecondsDoesNotComplete();
+    testToneConventionMatchesWsjtx();
+    testFrameEndsAreTapered();
+    testLateStartSkipsIntoTheFrame();
```
Comment (ten9876) [inline tests/wspr_beacon_test.cpp:386]: Nit, non-blocking. These three are excellent — I reverted the tone convention and forced `m_envelope` to `1.0f` and watched exactly the right assertions fail, so the "verified to have teeth" claim holds.

The gap is that all four new cases cover the *generator*, and the symptom the PR opens with — "it opens the band in USB, not DIGU, with a very wide open filter" — is fixed somewhere else entirely, in `reassertBeaconChannel()`, which has no coverage at all. Neither does the speech-chain borrow/restore. Right now the only thing pinning the reported bug is the on-air run.

The repo does have dialog-level test precedent (`settings_browser_dialog_test.cpp`, `copy_assist_settings_dialog_test.cpp`, `flex_control_dialog_size_test.cpp`), so the decision table in `reassertBeaconChannel()` — dial moved → abort; mode wrong → re-assert; transmitter busy → refuse — is reachable without a live radio. Not asking for it in this PR; noting it so the next regression in that function isn't found on the air.

File: src/gui/PskReporterMapDialog.cpp [hunk 6]
```diff
@@ -366,13 +371,39 @@ PskReporterMapDialog::PskReporterMapDialog(AudioEngine* audioEngine,
...
+        tr("Audio offset above the selected USB dial frequency, at the lowest "
+           "of the four tones — the same convention as WSJT-X"));
     beaconRow->addWidget(m_beaconTone);
 
+    beaconRow->addWidget(new QLabel(tr("Level:"), beaconBox));
+    m_beaconLevel = new QSpinBox(beaconBox);
+    // Hard-coded at -20 dBFS before this. WSJT-X generates at full scale and
+    // attenuates digitally through its Pwr slider (SoundOutput::setAttenuation);
+    // the equivalent knob here had no UI at all, so an operator who came out
+    // underdriven had nothing to reach for. Floor of -3 rather than 0 keeps a
```
Comment (ten9876) [inline src/gui/PskReporterMapDialog.cpp]: Nit: −3 is the ceiling of the range, not the floor.

```suggestion
    // underdriven had nothing to reach for. Ceiling of -3 rather than 0 keeps a
```

File: src/gui/PskReporterMapDialog.cpp [hunk 7]
```diff
@@ -366,13 +371,39 @@ PskReporterMapDialog::PskReporterMapDialog(AudioEngine* audioEngine,
...
+        tr("Audio tone from 1400 to 1600 hertz"));
+    // This is the frequency of the LOWEST of the four tones, which is what
+    // WSJT-X's Tx-frequency box means too. Naming it "center" (as this did)
+    // was not just loose wording — the generator centred the constellation on
+    // it, putting every spot 2.2 Hz below where the same number in WSJT-X
+    // would have put it.
     m_beaconTone->setToolTip(
-        tr("Audio offset above the selected USB dial frequency"));
+        tr("Audio offset above the selected USB dial frequency, at the lowest "
+           "of the four tones — the same convention as WSJT-X"));
```
Comment (ten9876) [inline src/gui/PskReporterMapDialog.cpp:392]: Nit, non-blocking, and explicitly *not* a request to change the range in this PR — just recording the consequence so it isn't rediscovered later.

Now that this value is tone 0 rather than the centre, the spin box maximum of 1600 puts symbol 3 at 1604.4 Hz, outside the 1400–1600 WSPR sub-band; the clean maximum under the new convention is 1595.6. The old centred convention overran the top too (1600 → 1597.8…1602.2), and it also overran the *bottom* (1400 → 1397.8), which the new convention fixes — so this is strictly better at one end and slightly worse at the other. Default 1500 → 1500…1504.4 is comfortably inside either way. Worth a follow-up on the spin box range, not a change here.

**Review by ten9876 — APPROVED** [review]
Approving. I pushed `ff62cc7b` to this branch resolving everything from my review and @aethersdr-agent's — the fixes are in the branch rather than left as asks, so this is a review of the code as it now stands.

## What changed

**The speech chain moved from arm to the key** (`borrowBeaconSpeechChain()`, called at the end of `reassertBeaconChannel()`). This is the one finding that was more than a nit, and the argument for it is the PR's own: `reassertBeaconChannel()` exists because "one push at arm time is not enough", and that reasoning was applied to mode and both passbands but not to the four station settings borrowed a few lines above them. Anything that turned the speech processor back on inside the armed window transmitted through it, and `m_beaconPrevSpeechProc` — latched at arm — handed back the wrong value afterwards. Same lead-in, same `isTransmitting()/isTuning()` guard, and the four writes precede `xmit 1` on the same ordered connection.

@jensenpat — your objection to the VOX argument was about using VOX *during* a WSPR transmission, and you're right about that. The complaint was about the *armed* window, which your own new deferral path in `updateBeaconState()` is written to accommodate ("MOX for a quick voice contact"). Moving the borrow resolves it without anyone having to concede the point: nothing is stripped from the station until the beacon actually keys, so there is no window left to argue about.

**The tail taper now wins over the head ramp.** A skip landing inside the tail region ramped *up* over the remaining frames and stopped dead — the exact click the taper removes. Unreachable from `updateBeaconState()`, reachable through the public `start()`. `testSkipIntoTheTailStillFadesOut()` pins it, and fails at 0.96 of full amplitude with the old ordering, not marginally.

**Docs and comments**: `pipelines.md` listed the borrowed state without VOX and described all of it as taken at arm; rewritten for what the code does. "Floor of -3" is the ceiling. And I recorded *why* the tone spin box keeps 1400–1600 rather than narrowing to 1595.6 — that is WSJT-X's own WSPR range under the same convention, so matching it is the whole point; the centred convention this replaces overran the top too, and also overran the bottom, which you fixed.

## Verified

Everything below was run locally, on a merge of this branch onto current `main` (27 commits ahead of the base this PR's CI ran against — the merge is clean, and main's `AudioEngine.cpp` churn does not touch the WSPR pump):

- Full app and all 221 test targets build; `wspr_beacon_test` green. Suite 219/221, both failures pre-existing and unrelated (`cross_needle_meter_test` is a local font-resolution artifact, `automation_drag_at_test` is timing-flaky under a parallel run; both pass on their own).
- **Your tests have teeth — I checked rather than took your word for it.** Restoring the centred tone convention fails both tone assertions; forcing `m_envelope` to `1.0f` fails all three taper assertions.
- `kTaperFrames` = 278, seed `0.9604^278` ≈ −97.6 dB, the ramp reaches unity at emitted frame 277 and the `else` branch pins it there, so drift across the divisions cannot leak into the body. Tail is exactly 278 frames.
- Lead-in arithmetic correct in both branches: on time, audio starts at slot + 1.000 s regardless of tick jitter; late, `skipFrames = (lateMs − 1000)·24` is exactly the message position at that wall time.
- The re-assert genuinely reaches the wire: neither `SliceModel::setFilterWidth()` nor `TransmitModel::setTxFilter()` has an equality guard, and while `SliceModel::setMode()` does, the status path (`SliceModel.cpp:1068`) mirrors the radio's mode back into `m_mode` — so a band-stack recall to USB leaves the model reading USB and the push happens.
- `setDexp` → `transmit set compander=` (`TransmitModel.cpp:579-585`) — right handle for the compander.

## Still open, deliberately

`reassertBeaconChannel()` has no automated coverage, so the symptom this PR opens with is pinned only by the on-air run. Closing that needs either a seam cut purely for testability or a test that waits on a real two-minute slot boundary; neither is a good trade against a change with this much on-air evidence behind it. Worth a follow-up issue rather than a blocker.

The on-air result is the part that carries this: +2.5 Hz measured by ~60 independent receivers against a +2.197 Hz prediction is evidence no unit test sharing the convention could have produced, and you were right to say so.

File: tests/wspr_beacon_test.cpp (hunk 5 above)
Comment (ten9876) [reply tests/wspr_beacon_test.cpp:386]: Resolving as a deliberate follow-up rather than as fixed: `reassertBeaconChannel()` still has no automated coverage. Closing it needs either a seam cut purely for testability or a test that waits on a real two-minute slot boundary, and neither looked like a good trade against a change with this much on-air evidence behind it. `ff62cc7b` does add `testSkipIntoTheTailStillFadesOut()` for the taper-ordering fix in that commit, so the generator side stays fully pinned.
